gui/src/shortcutdialog.cpp

```diff
@@ -0,0 +1,311 @@
+    if (access(chiakiFlatpakDir.c_str(), 0) == 0) {
+        return "flatpak run io.github.streetpea.Chiaki4deck";
+    }
+    else {
+        char buffer[PATH_MAX];
+        ssize_t len = readlink("/proc/self/exe", buffer, sizeof(buffer) - 1);
+
+        if (len != -1) {
+            buffer[len] = '\0';
+            return buffer;
+        }
+    }
+#endif
+    return "";
+}
+
+void ShortcutDialog::CreateShortcut(const DisplayServer* displayServer, std::map<std::string, std::string> artwork) {
+    //Create a map of variables for the template
+    std::map<std::string, std::string> paramMap;
+
+    paramMap["home_ssid"] = local_ssid_edit->text().toStdString();
+    paramMap["ps_console"] = displayServer->IsPS5() ? "5" : "4";
+    paramMap["wait_timeout"] = "35";
+    paramMap["home_addr"] = displayServer->GetHostAddr().toStdString();
+    paramMap["away_addr"] = external_dns_edit->text().toStdString();
+    paramMap["regist_key"] = QString::fromUtf8(displayServer->registered_host.GetRPRegistKey()).toStdString();
+    paramMap["login_passcode"] = passcode_edit->text().toStdString();
+    paramMap["mode"] = mode_combo_box->currentText().toStdString();
+    paramMap["server_nickname"] = displayServer->discovered ? displayServer->discovery_host.host_name.toStdString() : displayServer->registered_host.GetServerNickname().toStdString();
+    paramMap["executable"] = getExecutable();
+    std::stringstream escapeCommandStream;
+    escapeCommandStream << "printf %q " << "\"" << paramMap["server_nickname"] << "\"";
+    const std::string escapeCommandStr = escapeCommandStream.str();
+    const char* escapeCommand = escapeCommandStr.c_str();
+    std::array<char, 128> buffer;
+    std::string result;
+    std::unique_ptr<FILE, decltype(&pclose)> pipe(popen(escapeCommand, "r"), pclose);
+    if (!pipe) {
+        throw std::runtime_error("popen() failed!");
+    }
+    while (fgets(buffer.data(), buffer.size(), pipe.get()) != nullptr) {
+        paramMap["escaped_server_nickname"] += buffer.data();
+    }
+
+
+    std::string fileText = compileTemplate("shortcut.tmpl", paramMap);
+
+    if (allow_external_access_checkbox->isChecked()) {
+        if (displayServer->IsPS5()) {
+            fileText = fileText+compileTemplate("ps5_external.tmpl", paramMap);
+            fileText = fileText+compileTemplate("discover_wakeup.tmpl", paramMap);
+        } else {
+            fileText = fileText+compileTemplate("ps4_external.tmpl", paramMap);
+        }
+    } else {
+        fileText = fileText+compileTemplate("local.tmpl", paramMap);
+        fileText = fileText+compileTemplate("discover_wakeup.tmpl", paramMap);
+    }
+
+    fileText = fileText+compileTemplate("launch.tmpl", paramMap);
+
+    bool steamExists = SteamShortcutParser::steamExists();
+    std::string executable = getExecutable();
+
+    std::string filePath;
+    if (steamExists && executable.rfind("flatpak run io.github.streetpea.Chiaki4deck", 0) == 0) {
+        filePath.append(getenv("HOME"));
```

Comment:
This should always save it in `$HOME/.config/Chiaki` and then you don't need special case for flatpak.